project/src/js/lichess/friends.js

```diff
@@ -11,18 +43,29 @@ export default {
     return onlineFriends.length;
   },
 
-  set(list) {
-    onlineFriends = list;
+  set(friends, playings) {
+      onlineFriends = friends.map(name => makeUser(name, false));
+      for (var user of playings) setPlaying(user, true);
+      onlineFriends.sort(lexicallyCompareFriends);
+  },
+
+  add(name, playing) {
+    var friend = makeUser(name, playing);
+
+    onlineFriends.push(friend);
+    onlineFriends.sort(lexicallyCompareFriends);
+  },
+
+  playing(name) {
+    setPlaying(name, true);
   },
 
-  add(name) {
-    onlineFriends.push(name);
-    onlineFriends.sort(caseInsensitiveSort);
+  stopped_playing(name) {
+    setPlaying(name, false);
   },
 
-  remove(name) {
-    const i = onlineFriends.indexOf(name);
-    if (i !== -1) onlineFriends.splice(i, 1);
+  remove(leaving) {
+    onlineFriends.filter(friend => friend.name != leaving);
```

Comment:
`filter` doesn't mutate the array, so you need to do `onlineFriends = onlineFriends.filter(...)` for this to work